### ISSUE TITLE
Hoe can I update the existing message?

### ISSUE THREAD

**Issue by bushev** [FredrikOseberg/react-chatbot-kit#132]
Hi,

I am in the process of integrating this library with a WebSocket server that broadcasts messages partially. Consequently, whenever I receive an extension of a message (identified by a specific ID), it is necessary to update the corresponding message in the chat.

I previously attempted to substitute the entire message, but this occasionally caused the message to be divided into two to four separate segments.

<img width="294" alt="image" src="https://user-images.githubusercontent.com/7535766/225923925-b84e003a-2447-4823-b1ae-8b25c7e5797c.png">

**Comment by sam-h-bean** [comment]
Yeah would also like to use server sent events for this @FredrikOseberg

**Comment by zoetaka38** [comment]
@bushev 
I was able to do this by doing the following:
```
messageBuffer += new TextDecoder("utf-8").decode(value)
var botMessage = createChatBotMessage(messageBuffer)

setState((prev: any) => {
  const prevMessagesWithoutLast = prev.messages.slice(0, -1)
  return {
    ...prev,
    messages: [...prevMessagesWithoutLast, botMessage],
  }
})
```

in my situation, i got the response as streaming response from fastapi (python).

**Comment by OliverThomas2000** [comment]
Try this out: 
```
const updateLastMessage = (message) => {
    setState((prev) => {
      return { ...prev, messages: [...prev.messages.slice(0, -1), { ...prev.messages.at(-1), message }]};
    });
  };
  ```
